docs: inline autocomplete and TCG (#5263)
forcedotcom/salesforcedx-vscode: docs/_articles/en/einstein/einstein-overview.md

## Current Capabilities

> Use Einstein for Developers to generate boilerplate Apex code from natural language prompts and to automatically receive code completion suggestions inline. Used along with IntelliSense, this feature makes Apex development tooling in Visual Studio Code even richer. Familiarity with Visual Studio Code is assumed.

Reviewer: ```suggestion
Einstein for Developers generates Apex code from natural language prompts and automatically suggests code completions for you as you type. When enabled along with IntelliSense, this feature makes Apex development tooling in Visual Studio Code even richer. Familiarity with Visual Studio Code is assumed.
```
Can we make it seem more like E4D is _doing something for you_ rather than you are _using_ E4D?  I like thinking of Inline Completions like Einstein is a person/thing that knows what I need and is trying to help me out and anticipate what I want.

Enter natural language instructions in a sidebar, so you can work with your editor and the tool side by side, without any interruptions to your workflow. You can also get code suggestions within an existing Apex class, trigger, or anonymous Apex file. Use the VS Code Command Palette to enter a prompt describing what you'd like to build and then generate code suggestions within your editor.

Enable inline autocompletion to automatically receive suggestions as you code.

> **Note**: Einstein GPT for Developers uses generative AI, which can produce inaccurate or harmful responses. The output generated by AI is often nondeterministic. Before using the generated output, review it for accuracy and safety. You assume responsibility for how the outcomes of Einstein are applied to your organization.

Reviewer 2: ```suggestion
**Note**: Einstein for Developers uses generative AI, which can produce inaccurate or harmful responses. The output generated by AI is often nondeterministic. Before using the generated output, review it for accuracy and safety. You assume responsibility for how the outcomes of Einstein are applied to your organization.
```
